src/NzbDrone.Core/Download/ProcessDownloadDecisions.cs

```diff
@@ -36,52 +38,91 @@ public ProcessedDecisions ProcessDecisions(List<DownloadDecision> decisions)
             var grabbed = new List<DownloadDecision>();
             var pending = new List<DownloadDecision>();
+            var storedUsenet = new List<DownloadDecision>();
+            var fallbackUsenet = new List<DownloadDecision>();
+            var storedTorrent = new List<DownloadDecision>();
+            var fallbackTorrent = new List<DownloadDecision>();
 
             foreach (var report in prioritizedDecisions)
             {
                 var remoteEpisode = report.RemoteEpisode;
+                var downloadProtocol = report.RemoteEpisode.Release.DownloadProtocol;
 
-                var episodeIds = remoteEpisode.Episodes.Select(e => e.Id).ToList();
-
-                //Skip if already grabbed
-                if (grabbed.SelectMany(r => r.RemoteEpisode.Episodes)
-                                .Select(e => e.Id)
-                                .ToList()
-                                .Intersect(episodeIds)
-                                .Any())
+                // Skip if already grabbed
+                if (IsEpisodeProcessed(grabbed, report))
                 {
                     continue;
                 }
 
                 if (report.TemporarilyRejected)
                 {
-                    _pendingReleaseService.Add(report);
+                    _pendingReleaseService.Add(report, PendingReleaseReason.Delay);
                     pending.Add(report);
                     continue;
                 }
 
-                if (pending.SelectMany(r => r.RemoteEpisode.Episodes)
-                        .Select(e => e.Id)
-                        .ToList()
-                        .Intersect(episodeIds)
-                        .Any())
+                if (IsEpisodeProcessed(pending, report))
                 {
                     continue;
                 }
 
+                if (downloadProtocol == DownloadProtocol.Usenet)
+                {
+                    if (IsEpisodeProcessed(storedUsenet, report))
+                    {
+                        fallbackUsenet.Add(report);
+                        continue;
+                    }
+                    else if (storedUsenet.Any())
+                    {
+                        storedUsenet.Add(report);
+                        continue;
+                    }
+                }
+
+                if (downloadProtocol == DownloadProtocol.Torrent)
+                {
+                    if (IsEpisodeProcessed(storedTorrent, report))
+                    {
+                        fallbackTorrent.Add(report);
+                        continue;
+                    }
+                    else if (storedTorrent.Any())
+                    {
+                        storedTorrent.Add(report);
+                        continue;
+                    }
+                }
+
                 try
                 {
                     _downloadService.DownloadReport(remoteEpisode);
```

Comment:
I failed to realize, but I don't think we actually throw DownloadClientUnavailableException and simply return in side DownloadReport if no download client is enabled/available.